libsolidity/ast/Types.cpp

```diff
@@ -2335,11 +2335,13 @@ bool StructType::isDynamicallyEncoded() const
 {
 	if (recursive())
```

Comment:
While it might be correct due to assertions, it looks wrong. Whether or not it is dynamically encoded should be determined by the fact that it is a storage or a memory struct.

Reply:
Or actually put differently: recursive structs should not be encoded at all.